src/components/Login/index.tsx

```diff
@@ -1,94 +1,181 @@
-        }
-    }
+//TYPES for stateProps
+interface StateProps {
+  authentication: Reducer
+}
+
+//DispatchProps
+
+interface data {
+  status?: string,
+  message?: string,
+  Authorization?: string
+}
+
+interface LOG_IN {
+  type: typeof LOG_IN,
+  payload: LOGIN_STATE
+}
 
-    // TODO: integrate in LOGIN component
-    // componentDidMount(){
-    //   request('/auth/token')
-    //   .then(response => this.props.setAuthentication(response.data))
-    //   .catch(err => this.props.setAuthentication(null))
-    // }
-
-
-    handleSignIn = () =>{
-      //#TODO update url to meet api standard 
-      fetch("/auth", {
-        method: "post",
-        headers: {'Content-Type': 'application/json'},
-        body: JSON.stringify({
-          username: this.state.username,
-          password: this.state.password
-        })
-      }).then((response) => response.json())
-      .then((data) => {
-          // TODO: find what data backend sends us
-          if (!data.status) {
-            this.setState({ error : true });
-          } else {
-            // TODO: Update redux to call authenticate action as true
-          }
-      })
-      .catch((err) => {
-          console.log(err)
-          this.setState({ error : true });
-      })
-    
+interface DispatchProps {
+  logIn: (payload: LOGIN_STATE) => LOG_IN
+}
+
+
+type props = StateProps & DispatchProps & Props
+
+interface IErrorLoginState {
+  showErrorMessage: Boolean;
+  username: string;
+  password: string;
+  error: boolean;
+  redirectPage: boolean;
+}
+
+class Login extends React.Component<props, IErrorLoginState> {
+
+  constructor(props: props) {
+    super(props);
+    this.state = {
+      showErrorMessage: false,
+      username: '',
+      password: '',
+      error: false,
+      redirectPage: false
     }
+  }
 
-    render() {
-      // TODO: If redux authenticated is true: redirect to user dashboard
+  // TODO: integrate in LOGIN component
+  // componentDidMount(){
+  //   request('/auth/token')
+  //   .then(response => this.props.setAuthentication(response.data))
+  //   .catch(err => this.props.setAuthentication(null))
+  // }
 
+  componentDidMount = () => {
+    // this.handleSignIn()
+  }
+  
+  handleSignIn = async () => {
+    // this.props.logIn({username: this.state.username, password: this.state.password})
+    const body = {username: this.state.username, password: this.state.password}
+
+    return fetch(`https://calcumon-user-api.herokuapp.com/auth/login`, {
+    method: 'POST',
+    headers: {
+      'accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(body)
+  }).then(response=>{
+      return response.json()
+    }).then((data)=>{
+      // this.props.history.push('/Dashboard')
+      this.props.logIn(data)
+      console.log(data)
+      this.setState({ redirectPage: true })
+      this.props.loggedIn(true)
+    })
+    .catch(err=>{
+      console.log(err)
+      // this.props.LogIn(err)
```

Comment:
How are we handling errors currently? Maybe a login error message under the login form, prompting users to try again?